mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/core/CleanupOnCloseRpcSystem.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.server.core;
+
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.nio.file.Path;
+import javax.annotation.Nullable;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.core.classloading.ComponentClassLoader;
+import org.apache.flink.runtime.rpc.AddressResolution;
+import org.apache.flink.runtime.rpc.RpcSystem;
+import org.apache.flink.util.FileUtils;
+import org.apache.flink.util.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** An {@link RpcSystem} wrapper that cleans up resources after the RPC system has been closed. */
+class CleanupOnCloseRpcSystem implements RpcSystem {
+    private static final Logger LOG = LoggerFactory.getLogger(CleanupOnCloseRpcSystem.class);
+
+    private final RpcSystem rpcSystem;
+    private final ComponentClassLoader pluginLoader;
+    @Nullable private final Path tempDirectory;
+
+    public CleanupOnCloseRpcSystem(
+            RpcSystem rpcSystem, ComponentClassLoader pluginLoader, @Nullable Path tempDirectory) {
+        this.rpcSystem = Preconditions.checkNotNull(rpcSystem);
+        this.pluginLoader = Preconditions.checkNotNull(pluginLoader);
+        this.tempDirectory = tempDirectory;
+    }
+
+    @Override
+    public void close() {
```

Comment:
do we actually utilize the close anywhere?

Reply:
flink-rpc-akka seems to need this.